packager/mpd/base/xml/xml_node.cc

```diff
@@ -460,14 +460,43 @@ bool RepresentationXmlNode::AddAudioChannelInfo(const AudioInfo& audio_info) {
   std::string audio_channel_config_value;
 
   if (audio_info.codec() == kEC3Codec) {
+    const auto& codec_data = audio_info.codec_specific_data();
     // Convert EC3 channel map into string of hexadecimal digits. Spec: DASH-IF
     // Interoperability Points v3.0 9.2.1.2.
     const uint16_t ec3_channel_map =
-        base::HostToNet16(audio_info.codec_specific_data().ec3_channel_map());
-    audio_channel_config_value =
+      base::HostToNet16(codec_data.ec3_channel_map());
+    // Calculate EC3 channel configuration descriptor value with MPEG scheme.
+    // Spec: ETSI TS 102 366 V1.4.1 Digital Audio Compression
+    // (AC-3, Enhanced AC-3) I.1.2.
+    const uint32_t ec3_channel_mpeg_value = codec_data.ec3_channel_mpeg_value();
+    const uint32_t NO_MAPPING = 0xFFFFFFFF;
+    if (ec3_channel_mpeg_value == NO_MAPPING) {
+      audio_channel_config_value =
         base::HexEncode(&ec3_channel_map, sizeof(ec3_channel_map));
-    audio_channel_config_scheme =
+      audio_channel_config_scheme =
         "tag:dolby.com,2014:dash:audio_channel_configuration:2011";
+    } else {
+      audio_channel_config_value = base::UintToString(ec3_channel_mpeg_value);
+      audio_channel_config_scheme = "urn:mpeg:mpegB:cicp:ChannelConfiguration";
+    }
```

Comment:
In the latest DASH-IF-IOP: https://dashif.org/docs/DASH-IF-IOP-v4.3.pdf

Section 3.9.4.6. It says:

> If the codec is anyone in Table 22, Table 27, Table 28 or Table 29, then any of the following may be used
> - urn:mpeg: dash:23003:3:audio_channel_configuration:2011 as defined in ISO/IEC 23009-1 [1], 5.8.5.4
> - urn:mpeg:mpegB:cicp:ChannelConfiguration as defined in ISO/IEC 23001-8 [49]
>
> If the codec is ec-3 or ac-4 according to Table 25, then the following shall be used
> - tag:dolby.com,2014: dash:audio_channel_configuration:2011" as defined at http://dashif.org/identifiers/audio-source-data/ (see section 9.2.1.2)

Table 25 is for Dolby EC-3, mlpa and AC-4
Table 22,27,28 is for MPEG-4 AAC; table 29 is for MPEG-H audio.

So per DASH-IF-IOP, we should be using "tag:dolby.com,2014: dash:audio_channel_configuration:2011" instead of "urn:mpeg:mpegB:cicp:ChannelConfiguration" for EC-3. Is there any change in Dolby's internal guidance on this?

Reply:
Yes. Dolby has changed it. Not only for internal guidance, but also try to fix in DASH-IF IOP (still under review). https://github.com/Dash-Industry-Forum/DASH-IF-IOP/issues/268